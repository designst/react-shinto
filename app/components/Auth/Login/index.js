/**
 * Login
 */

import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { FormattedMessage } from 'react-intl';

import Button from '@material-ui/core/Button';

import LoginForm from './form';
import messages from './messages';

const Login = ({ onSubmit }) => (
  <div className="b-login">
    <FormattedMessage {...messages.header} />

    <LoginForm onSubmit={onSubmit} />

    <Link to="/auth/register">
      <Button type="link">Register</Button>
    </Link>

    <Link to="/auth/password/reset">
      <Button type="link">Password Reset</Button>
    </Link>
  </div>
);

Login.propTypes = {
  onSubmit: PropTypes.func.isRequired,
};

export default Login;
