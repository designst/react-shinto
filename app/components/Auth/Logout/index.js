/**
 * Logout
 */

import React from 'react';
import PropTypes from 'prop-types';

class Logout extends React.Component {
  componentDidMount() {
    const { onSubmit } = this.props;

    onSubmit();
  }

  render() {
    return <div className="b-logout">Logout...</div>;
  }
}

Logout.propTypes = {
  onSubmit: PropTypes.func.isRequired,
};

export default Logout;
