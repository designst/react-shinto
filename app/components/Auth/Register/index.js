/**
 * Register
 */

import React from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';

import RegisterForm from './form';
import messages from './messages';

const Register = ({ onSubmit }) => (
  <div className="b-register">
    <FormattedMessage {...messages.header} />

    <RegisterForm onSubmit={onSubmit} />
  </div>
);

Register.propTypes = {
  onSubmit: PropTypes.func.isRequired,
};

export default Register;
