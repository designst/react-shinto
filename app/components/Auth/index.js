import React from 'react';
import PropTypes from 'prop-types';

import { renderRoutes } from 'react-router-config';

import Grid from '@material-ui/core/Grid';
import Paper from '@material-ui/core/Paper';
import { withStyles } from '@material-ui/core/styles';

import styles from './styles';

const Auth = ({ route, classes }) => (
  <Grid container justify="center" alignItems="center">
    <Grid item xs={12} sm={6} md={4} lg={3}>
      <Paper className={classes.root}>{renderRoutes(route.routes)}</Paper>
    </Grid>
  </Grid>
);

Auth.propTypes = {
  route: PropTypes.object.isRequired,
  classes: PropTypes.object.isRequired,
};

export default withStyles(styles)(Auth);
