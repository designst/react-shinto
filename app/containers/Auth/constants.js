import createConstant from 'utils/createConstant';

export const AUTH_API_REQUEST = createConstant(__filename, 'AUTH_API_REQUEST');
export const AUTH_API_SUCCESS = createConstant(__filename, 'AUTH_API_SUCCESS');
export const AUTH_API_FAILURE = createConstant(__filename, 'AUTH_API_FAILURE');
