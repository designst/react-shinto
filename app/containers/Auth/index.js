import { compose } from 'redux';
import { connect } from 'react-redux';

import Auth from 'components/Auth';

import injectReducer from 'utils/injectReducer';

import reducer from './reducer';

const mapStateToProps = () => ({});

const mapDispatchToProps = {};

const withReducer = injectReducer({ key: 'auth', reducer });

const withConnect = connect(mapStateToProps, mapDispatchToProps);

export default compose(withReducer, withConnect)(Auth);
