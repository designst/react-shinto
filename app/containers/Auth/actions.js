import { ERROR_ACTION, WAIT_FOR_ACTION } from 'redux-wait-for-action';

import { CHECK_AUTH_REQUEST, CHECK_AUTH_SUCCESS, CHECK_AUTH_FAILURE } from './constants';

export const checkAuthRequest = () => ({
  type: CHECK_AUTH_REQUEST,
});

export const checkAuthSuccess = data => ({
  type: CHECK_AUTH_SUCCESS,
  data,
});

export const checkAuthFailure = error => ({
  type: CHECK_AUTH_FAILURE,
  error,
});

export const checkAuthRequestWait = () => ({
  ...checkAuthRequest(),
  [WAIT_FOR_ACTION]: CHECK_AUTH_SUCCESS,
  [ERROR_ACTION]: CHECK_AUTH_FAILURE,
});
