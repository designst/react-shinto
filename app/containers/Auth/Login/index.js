import { compose } from 'redux';
import { connect } from 'react-redux';

import Login from 'components/Auth/Login';

import { loginRequest } from './actions';

const mapStateToProps = () => ({});

const mapDispatchToProps = {
  onSubmit: loginRequest,
};

const connector = connect(
  mapStateToProps,
  mapDispatchToProps,
);

export default compose(connector)(Login);
