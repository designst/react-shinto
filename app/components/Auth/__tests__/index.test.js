import React from 'react';
import renderer from 'react-test-renderer';
import { StaticRouter } from 'react-router-dom';

import Auth from '../index';

describe('<Auth />', () => {
  test('renders', () => {
    const mockData = {
      route: {},
    };

    const tree = renderer
      .create(
        <StaticRouter context={{}}>
          <Auth {...mockData} />
        </StaticRouter>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
