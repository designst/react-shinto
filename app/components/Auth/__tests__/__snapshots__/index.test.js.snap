// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Auth /> renders 1`] = `
<div
  className="MuiGrid-container-2 MuiGrid-align-items-xs-center-10 MuiGrid-justify-xs-center-19"
>
  <div
    className="MuiGrid-item-3"
  >
    <div
      className="MuiPaper-root-92 MuiPaper-elevation2-96 MuiPaper-rounded-93 Auth-root-1"
    />
  </div>
</div>
`;
